server/schemas/resolvers.js

```diff
@@ -1,31 +1,136 @@
+
+        // Resolver to fetch a single user by username or email
+        // ==================================================================
+        user: async (parent, { username, email }) => {
+            // Determine search criteria based on provided arguments
+            const params = username ? { username } : { email };
+            return await User.findOne(params);
+        },
+        // ==================================================================
+
+        // Resolver to fetch the currently authenticated user
+        // ==================================================================
+        me: async (parent, args, context) => {
+            // Check if user is authenticated
+            if (context.user) {
+                // Extract username from the user context
+                const { username } = context.user;
+                // Find the user and return their data, excluding the password
+                return await User.findOne({ username }).select('-password');
+            }
+            throw AuthenticationError;
+        },
+        // ==================================================================
+    },
+    // ==================================================================
+
+    // Resolvers for Mutations
+    // ==================================================================
+    Mutation: {
+        // Resolver to add a new user
+        // ==================================================================
+        addUser: async (parent, { username, email, password }) => {
+            // Create a new user with the provided credentials
+            const user = await User.create({ username, email, password });
+            // Sign a JWT token for the new user
+            const token = signToken(user);
+            // Return the token and user details
+            return { token, user };
+        },
+        // ==================================================================
+
+        // Resolver for user login
+        // ==================================================================
+        login: async (parent, { email, password }) => {
+            // Find the user by email
+            const user = await User.findOne({ email });
+            if (!user) {
+                throw AuthenticationError;
+            }
+            
+            // Check if the provided password is correct
+            const correctPw = await user.isCorrectPassword(password);
+
+            if (!correctPw) {
+                throw AuthenticationError;
+            }
+
+            const token = signToken(user);
+            return { token, user };
+        },
+        // ==================================================================
+
+        // Resolver to add preferences to a user
+        // ==================================================================
+        addPreferences: async (parent, { username, preferences }, context) => {
```

Comment:
I'm not sure how you are seeing the preferences, as a separate model which has a relation with user or a dependent of the user model